modules/osdk-crds.adoc

```diff
@@ -0,0 +1,201 @@
+which UI component to use. See the *openshift/console* project for a canonical
+link:https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L5-L26[list of React UI X-Descriptors] for {product-title}.
+
+Also see the *openshift/console* project for more information on
+link:https://github.com/openshift/console/tree/master/frontend/public/components/operator-lifecycle-manager/descriptors[Descriptors]
+in general.
+|===
+
+The following example depicts a `MongoDB Standalone` CRD that requires some user
+input in the form of a Secret and ConfigMap, and orchestrates Services,
+StatefulSets, Pods and ConfigMaps:
+
+[id="osdk-crds-owned-example_{context}"]
+.Example owned CRD
+[source,yaml]
+----
+      - displayName: MongoDB Standalone
+        group: mongodb.com
+        kind: MongoDbStandalone
+        name: mongodbstandalones.mongodb.com
+        resources:
+          - kind: Service
+            name: ''
+            version: v1
+          - kind: StatefulSet
+            name: ''
+            version: v1beta2
+          - kind: Pod
+            name: ''
+            version: v1
+          - kind: ConfigMap
+            name: ''
+            version: v1
+        specDescriptors:
+          - description: Credentials for Ops Manager or Cloud Manager.
+            displayName: Credentials
+            path: credentials
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:selector:core:v1:Secret'
+          - description: Project this deployment belongs to.
+            displayName: Project
+            path: project
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:selector:core:v1:ConfigMap'
+          - description: MongoDB version to be installed.
+            displayName: Version
+            path: version
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:label'
+        statusDescriptors:
+          - description: The status of each of the Pods for the MongoDB cluster.
+            displayName: Pod Status
+            path: pods
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:podStatuses'
+        version: v1
+        description: >-
+          MongoDB Deployment consisting of only one host. No replication of
+          data.
+----
+
+[id="osdk-crds-required_{context}"]
+== Required CRDs
+
+Relying on other required CRDs is completely optional and only exists to reduce
+the scope of individual Operators and provide a way to compose multiple
+Operators together to solve an end-to-end use case.
+
+An example of this is an Operator that might set up an application and install
+an etcd cluster (from an etcd Operator) to use for distributed locking and a
+Postgres database (from a Postgres Operator) for data storage.
+
+The Operator Lifecycle Manager (OLM) checks against the available CRDs and
```

Comment:
@ecordell OLM will resolve and install the `required` CRD, right? Do we need to point out it here?

Reply:
@ecordell ^^

Reply:
This is describing the "single-operator" workflow. If you have no catalogs and no subscriptions, OLM will still check that your required CRDs exist (if, for example, you have installed it outside of OLM)